oidc-controller/api/core/oidc/provider.py

```diff
@@ -25,35 +25,51 @@
 signing_key = RSAKey(key=rsa_load("signing_key.pem"), use="sig", alg="RS256")
 signing_keys = KEYS().append(signing_key)
 
-# TO BE REVIEWED
+# config from vc-authn-oidc 1.0 can be found here https://toip-vc-authn-controller-dev.apps.silver.devops.gov.bc.ca/.well-known/openid-configuration
+
 configuration_information = {
     "issuer": issuer_url,
     "authorization_endpoint": f"{issuer_url}/authorization",
     "token_endpoint": f"{issuer_url}/token",
     "jwks_uri": f"{issuer_url}/.well-known/openid-configuration/jwks",
-    "response_types_supported": ["code", "id_token token"],
+    "response_types_supported": ["code", "id_token", "token"],
     "id_token_signing_alg_values_supported": [signing_key.alg],
     "response_modes_supported": ["fragment", "query"],
     "subject_types_supported": ["public", "pairwise"],
-    "grant_types_supported": ["authorization_code", "implicit"],
+    "grant_types_supported": ["hybrid"],
     "claim_types_supported": ["normal"],
     "claims_parameter_supported": True,
     "claims_supported": ["sub"],
     "request_parameter_supported": True,
     "request_uri_parameter_supported": False,
     "scopes_supported": ["openid", "profile"],
+    "token_endpoint_auth_methods_supported": ["client_secret_basic"],
 }
 
 subject_id_factory = HashBasedSubjectIdentifierFactory("asdwadwa")
-authz_state = AuthorizationState(
-    subject_id_factory,
-    RedisWrapper(db_uri, db_name="provider", collection="authz_codes"),
-    RedisWrapper(db_uri, db_name="provider", collection="access_tokens"),
-    RedisWrapper(db_uri, db_name="provider", collection="refresh_tokens"),
-    RedisWrapper(db_uri, db_name="provider", collection="subject_identifiers"),
-)
-client_db = RedisWrapper(db_uri, db_name="provider", collection="clients")
-user_db = RedisWrapper(db_uri, db_name="provider", collection="users")
+
+
+kc_client = {
+    "enabled": True,
+    "client_id": "keycloak",
+    "client_name": "keycloak",
+    "allowed_grant_types": ["implicit", "code"],
+    "allowed_scopes": ["openid", "profile", "vc_authn"],
+    "response_types": ["code", "id_token", "token"],
+    "redirect_uris": [
+        "http://localhost:8880/auth/realms/vc-authn/broker/vc-authn/endpoint"
+    ],
+    "token_endpoint_auth_method": "client_secret_basic",
+    "require_client_secret": True,
+    "client_secret": "12345678",
+    "require_consent": False,
+}
+
+
 provider = Provider(
-    signing_key, configuration_information, authz_state, client_db, Userinfo(user_db)
+    signing_key,
+    configuration_information,
+    AuthorizationState(subject_id_factory),
+    {"keycloak": kc_client},
+    Userinfo({"Jason": {"sub": "Jason"}}),
 )
```

Comment:
This piece will be configuration-driven in the future, correct?

Reply:
yes, but I'm not sure about the user collection. It expects it to be a stored collection somewhere, so we may need a 'dummy' user record for the library... which is currently how this works.

Reply:
yes, you are right. Not sure how to manage the 'User' collection, as it wants to be a dataset, but in our use case it is not, so we may need a single 'dummy record' for the library's sake.